### PR TITLE
 [#4051] Use new JSON editor 

### LINKED ISSUE #4051: Use the Monaco Editor for the JSON Edit component
The admin currently uses a basic editor for JSON content. Using the [Monaco Editor](https://github.com/microsoft/monaco-editor/) will greatly improve dealing with JSON (folding, auto-closing brackets/quotes, etc).

There are two implementations of the editor in React:

https://github.com/suren-atoyan/monaco-react
https://github.com/react-monaco-editor/react-monaco-editor

The second one requires some webpack plugin, and I couldn't make it work. The first one does seem to work fine, so it should be used.

### LINKED ISSUE #4424: Migrate admin JSON components to Monaco editor
### Thema / Theme

Admin

### Omschrijving / Description

Following https://github.com/open-formulieren/open-forms/issues/4051, there are still places where the Monaco Editor can be used in the admin, but requires more work regarding styling.

In order of priority:

* [x] Form logic rules: Advanced JSON-logic trigger
* [x] Form logic rules: Assign variable-action
* [x] Variables tab: Default value for objects and arrays
* [x] Service fetch logic expression
* [x] Form logic rules: Show DSL

### Added value / Toegevoegde waarde

_No response_

### Aanvullende opmerkingen / Additional context

_No response_

### PR DESCRIPTION
**Pull request by Viicos**
Closes #4424

**Changes**

Update builder
Use new editor in various places

**Checklist**

Check off the items that are completed or not relevant.

- Impact on features

  - [x] Checked copying a form
  - [x] Checked import/export of a form
  - [x] Config checks in the configuration overview admin page
  - [x] Problem detection in the admin email digest is handled

- Release management

  - [x] I have labelled the PR as "needs-backport" accordingly

- I have updated the translations assets (you do NOT need to provide translations)

  - [x] Ran `./bin/makemessages_js.sh`
  - [x] Ran `./bin/compilemessages_js.sh`

- Commit hygiene

  - [x] Commit messages refer to the relevant Github issue
  - [x] Commit messages explain the "why" of change, not the how

### CODE REVIEW

File: src/openforms/js/components/formio_builder/WebformBuilder.js [hunk 1]
```diff
@@ -159,6 +160,7 @@ class WebformBuilder extends WebformBuilderFormio {
             // Context binding
             uniquifyKey={uniquifyKey}
             supportedLanguageCodes={LANGUAGES}
+            theme={currentTheme.value}
```
Comment (Viicos) [inline src/openforms/js/components/formio_builder/WebformBuilder.js:163]: I can't use the `useGlobalState` hook here, but I think this is ok as the light/dark mode can't be toggled when the component editor is opened. Closing the editor, toggling light/dark mode and opening the editor again works as expected.

**Comment by Viicos** [comment]
Not having the ability to specify rows/columns is very inconvenient, I wonder if we should simply support `cols`/`rows` props and multiply them by a fixed amount of pixels (i.e. 19px for height) to then get the appropriate CSS `height`/`width`

File: src/openforms/js/compiled-lang/en.json
Comment (sergei-maertens) [inline src/openforms/js/compiled-lang/en.json:1]: I need to check what's happening here :/

File: src/openforms/js/components/admin/form_design/logic/Trigger.js [hunk 2]
```diff
@@ -313,10 +316,13 @@ const Trigger = ({name, logic, onChange, error, withDSLPreview = false, children
   // parent component
   useOnChanged(jsonLogicFromState, () => onChange({target: {name, value: jsonLogicFromState}}));
 
-  return (
-    <div className="logic-trigger">
-      <div className="logic-trigger__editor">
-        {error && <div className="logic-trigger__error">{error}</div>}
+  {
+    /* TODO: ensure this is rendered properly for Price logic */
+  }
```
Comment (sergei-maertens) [inline src/openforms/js/components/admin/form_design/logic/Trigger.js:321]: Need to double check and clean up this comment

File: src/openforms/forms/tests/e2e_tests/helpers.py [hunk 3]
```diff
@@ -30,3 +33,85 @@ async def click_modal_button(page: Page, button_text: str, **kwargs):
...
+
+    # 1. using page.keyboard.insert_text doesn't work, because the editor automatically
+    #    inserts matching quotes and curly braces, which causes broken syntax...
+    # 2. using navigator.clipboard.writeText(text) directly doesn't work because of
+    #    permission issues with webkit, see
+    #    https://github.com/microsoft/playwright/issues/13037
+    #
+    # Instead, use a dummy text area and interact with it to CTRL + C, CTRL + V
+    code = json.dumps(expression)
+    await editor.evaluate(
```
Comment (sergei-maertens) [inline src/openforms/forms/tests/e2e_tests/helpers.py:64]: need to give https://github.com/microsoft/playwright/issues/8114 another try...

File: src/openforms/js/components/admin/form_design/variables/ServiceFetchConfigurationForm.js [hunk 4]
```diff
@@ -222,10 +222,10 @@ const ServiceFetchConfigurationForm = ({formik, selectExisting = false}) => {
                 >
                   <JsonWidget
                     name="body"
-                    id="id_body"
                     logic={formik.values.body}
                     cols={20}
```
Comment (sergei-maertens) [inline src/openforms/js/components/admin/form_design/variables/ServiceFetchConfigurationForm.js:226]: need to make this 50

File: src/openforms/js/components/admin/form_design/variables/ServiceFetchConfigurationForm.js [hunk 5]
```diff
@@ -278,7 +278,8 @@ const ServiceFetchConfigurationForm = ({formik, selectExisting = false}) => {
                     name="jsonLogicExpression"
                     logic={formik.values.jsonLogicExpression || ''}
                     cols={20}
```
Comment (sergei-maertens) [inline src/openforms/js/components/admin/form_design/variables/ServiceFetchConfigurationForm.js:280]:
```suggestion
                    cols={50}
```

**Pull request merged**